File: src/arrays/max_sub_array.py
from typing import List


class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]
        current_sum = 0
        for el in nums:
            if current_sum < 0:
                current_sum = 0
            current_sum += el
            max_sum = max(current_sum, max_sum)
        return max_sum

File: java/src/main/java/com/blind75/arrays/MaxSubArray.java
package com.blind75.arrays;

public class MaxSubArray {

  public int maxSubArray(int[] nums) {
    var maxSum = nums[0];
    var currentSum = 0;
    for (int num : nums) {
      if (currentSum < 0) {
        currentSum = 0;
      }
      currentSum += num;
      maxSum = Math.max(maxSum, currentSum);
    }
    return maxSum;
  }

}
